tools/toolchain-tester/tests.py

```diff
@@ -97,8 +97,12 @@ def handle_expecteds(self, res: subprocess.CompletedProcess):
                     failing_test=self,
                 )
 
-        if expected.get("abi"):
-            expected_abi = expected["abi"]
+        if expected.get("abi") or expected.get("abi-file"):
+            if expected.get("abi"):
+                expected_abi = expected["abi"]
+            else:
+                expected_abi_file = open(expected["abi-file"])
+                expected_abi = expected_abi_file.read()
```

Comment:
This needs either a `expected_abi_file.close()` or

```
with open(expected["abi-file"]) as expected_abi_file:
   expected_abi = ...
```

Reply:
gotcha, thanks for catching that.